src/System.Text.Encoding/tests/EncodingTestHelpers.cs

```diff
@@ -0,0 +1,229 @@
+
+            // CharIndex < 0 or >= chars.Length
+            Assert.Throws<ArgumentOutOfRangeException>("charIndex", () => encoding.GetChars(new byte[4], 0, 4, new char[1], -1));
+            Assert.Throws<ArgumentOutOfRangeException>("charIndex", () => encoding.GetChars(new byte[4], 0, 4, new char[1], 2));
+
+            // Chars does not have enough capacity to accomodate result
+            Assert.Throws<ArgumentException>("chars", () => encoding.GetChars(new byte[4], 0, 4, new char[1], 1));
+        }
+
+        public static void GetMaxByteCount_Invalid(Encoding encoding)
+        {
+            Assert.Throws<ArgumentOutOfRangeException>("charCount", () => encoding.GetMaxByteCount(-1));
+            Assert.Throws<ArgumentOutOfRangeException>("charCount", () => encoding.GetMaxByteCount(int.MaxValue / 2));
+            Assert.Throws<ArgumentOutOfRangeException>("charCount", () => encoding.GetMaxByteCount(int.MaxValue));
+        }
+
+        public static void GetMaxCharCount_Invalid(Encoding encoding)
+        {
+            Assert.Throws<ArgumentOutOfRangeException>("byteCount", () => encoding.GetMaxCharCount(-1));
+        }
+
+        public static void GetString(Encoding encoding, byte[] bytes, int index, int count, string expected)
+        {
+            if (index == 0 && count == bytes.Length)
+            {
+                // Use GetString(byte[])
+                Assert.Equal(expected, encoding.GetString(bytes));
+            }
+            // Use GetString(byte[], int, int)
+            Assert.Equal(expected, encoding.GetString(bytes, index, count));
+        }
+
+        public static void GetString_Invalid(Encoding encoding)
+        {
+            // Bytes is null
+            Assert.Throws<ArgumentNullException>("bytes", () => encoding.GetString(null));
+            Assert.Throws<ArgumentNullException>("bytes", () => encoding.GetString(null, 0, 0));
+            
+            // Index or count < 0
+            Assert.Throws<ArgumentOutOfRangeException>("index", () => encoding.GetString(new byte[1], -1, 0));
+            Assert.Throws<ArgumentOutOfRangeException>("count", () => encoding.GetString(new byte[1], 0, -1));
+
+            // Index + count > bytes.Length
+            Assert.Throws<ArgumentOutOfRangeException>("bytes", () => encoding.GetString(new byte[1], 2, 0));
+            Assert.Throws<ArgumentOutOfRangeException>("bytes", () => encoding.GetString(new byte[1], 1, 1));
+            Assert.Throws<ArgumentOutOfRangeException>("bytes", () => encoding.GetString(new byte[1], 0, 2));
+        }
+
+        public static string GetString() => GetString(2, 260);
+        public static string GetString(int length) => GetString(length, length);
+        public static string GetString(int minLength, int maxLength) => s_randomGenerator.GetString(-55, false, minLength, maxLength);
+
+        public static string GetUnicodeString(int length)
```

Comment:
I'm confused as to what you mean here. Do you mean rename this helper method to "GetRandomString", or replace all usage of this metod with `GetString(int length)` from above?